third-party/build-gcc.sh

```diff
@@ -159,38 +153,76 @@ function setup_deps {
-                --enable-threads=posix                          \
-                --enable-__cxa_atexit                           \
-                --enable-clocale=gnu                            \
-                --enable-languages=c,c++                        \
-                --enable-lto                                    \
-                --disable-bootstrap                              \
-                --disable-nls                                   \
-                --disable-multilib                              \
-                --disable-install-libiberty                     \
-                --disable-werror                                \
-                --with-system-zlib
-                #--host=x86_64-vesoft-linux                                \
-                #--build=x86_64-vesoft-linux                                \
+    mkdir -p $gcc_object_dir
+    cd $gcc_object_dir
+    $source_dir/gcc-$gcc_version/configure      \
+        --prefix=$install_dir                   \
+        --with-pkgversion="Nebula Graph Build"  \
+        --enable-shared                         \
+        --enable-threads=posix                  \
+        --enable-__cxa_atexit                   \
+        --enable-clocale=gnu                    \
+        --enable-languages=c,c++                \
+        --enable-lto                            \
+        --enable-bootstrap                      \
+        --disable-nls                           \
+        --disable-multilib                      \
+        --disable-install-libiberty             \
+        --with-system-zlib                      \
+        --build=$triplet                        \
+        --host=$triplet                         \
+        --host=$triplet                         \
+        --disable-werror
     [[ $? -eq 0 ]] || exit 1
     cd $OLDPWD
 }
 
 function build_gcc {
-    cd $object_dir
-    make -j 20  |& tee build.log
+    cd $gcc_object_dir
+    make -s -j 20  |& tee build.log
     [[ $? -ne 0 ]] && exit 1
     cd $OLDPWD
 }
 
 function install_gcc {
-    cd $object_dir
-    make install-strip
+    cd $gcc_object_dir
+    make -s -j8 install-strip
+    [[ $? -ne 0 ]] && exit 1
+    cd $OLDPWD
+}
+
+function configure_binutils {
+    mkdir -p $bu_object_dir
+    cd $bu_object_dir
+    $source_dir/binutils-$bu_version/configure  \
+        --prefix=$install_dir                   \
+        --with-pkgversion="Nebula Graph Build"  \
+        --disable-shared                        \
+        --disable-nls                           \
+        --enable-gold                           \
+        --enable-ld=default                     \
+        --with-system-zlib                      \
+        --build=$triplet                        \
+        --host=$triplet                         \
+        --host=$triplet                         \
+        --disable-werror
+
+    cd $OLDPWD
+}
+
+function build_binutils {
+    cd $bu_object_dir
```

Comment:
Why build it independently now?

Reply:
Because it is problematic when building with a ancient compiler, like GCC 4.4.4.